constraint/pkg/apis/constraints/apis.go

```diff
@@ -17,6 +17,17 @@ const (
 	//
 	// This is the default EnforcementAction.
 	EnforcementActionDeny = "deny"
+
+	EnforcementActionScoped = "scoped"
+
+	// WebhookEnforcementPoint is the enforcement point for admission.
+	WebhookEnforcementPoint = "validation.k8s.io"
+
+	// AuditEnforcementPoint is the enforcement point for audit.
+	AuditEnforcementPoint = "audit.gatekeeper.sh"
+
+	// GatorEnforcementPoint is the enforcement point for gator cli.
+	GatorEnforcementPoint = "gator.gatekeeper.sh"
```

Comment:
Only `WebhookEnforcementPoint` constant is used to retrieve `enforcementAction` to create respective VAP with - https://github.com/open-policy-agent/frameworks/pull/403/files#diff-ac56566503d2145ac94cc1edbd630db0a048436f9c916956fd9ea35245636989R84 - audit and gator ep constants are only used in tests. So it is possible to have custom eps while integrating CF.

Reply:
For the VAP if we would need one predefined EP to properly create VAPB from constraint.

Reply:
One use for EP having strong types is if we plan to do validations against allowed actions for each EP. e.g. Deny, warn actions for the audit EP doesn't really make sense.

Reply:
But then `audit` action with `gator` or `gk-webhook` enforcementPoint doesnt match up.

Reply:
IMO audit is definitely an EP.

IIRC we suggested that EPs could do an analysis and report on the status of the constraint (or whatever status reporting mechanism the EP has) if there was some kind of invalid enforcement action?

The problem with Validating all known EPs or all acceptable actions for EPs are:

- CF can only hard code that logic for EPs it knows about, which limits its ability to be re-used
- Validating that all EPs are recognized assumes that the thing doing the validating knows about all available EPs. This may not be true (e.g. Gatekeeper knows nothing about terraform validator)
- An EP validating that it understands the actions it itself is intended to take makes total sense

IMO I'd avoid creating these constants here, since it implies that they can be used for things like validation. Definitely makes sense to define a special EP inside the VAP driver (whether that is  admission.k8s.io or something else).

Interesting side note: this dovetails nicely with the idea that the VAP driver probably should live in G8r rather than in the CF (that idea came up in other PRs)